raft/raft.cpp

```diff
@@ -1184,3 +1174,14 @@ raft::get_status()
 
     return status;
 }
+
+
+void
+raft::bail_on_exceeded_max_storage(bool do_throw)
+{
+    if (this->raft_log->maximum_storage_exceeded())
+    {
+        LOG(error) << MSG_ERROR_MAXIMUM_STORAGE_EXCEEDED;
```

Comment:
this->do_throw

Reply:
I wanted to leave one as a throw simply because I was able to unit test the functionality easily.

Reply:
Can we catch a SIGINT? Maybe with signal?